Sample with Servo engine doesn't work

Current master, trying to run sample-browser with Servo on ARM

The reason is ServoEngine.kt:35

> Process: org.mozilla.samples.browser, PID: 7616
>     java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.samples.browser/org.mozilla.samples.browser.BrowserActivity}: java.lang.UnsupportedOperationException: Not supported by this implementation:
>                 Use EngineSession.settings instead
>         at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
>         at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
>         at android.app.ActivityThread.-wrap11(Unknown Source:0)
>         at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
>         at android.os.Handler.dispatchMessage(Handler.java:106)
>         at android.os.Looper.loop(Looper.java:164)
>         at android.app.ActivityThread.main(ActivityThread.java:6494)
>         at java.lang.reflect.Method.invoke(Native Method)
>         at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
>         at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
>      Caused by: java.lang.UnsupportedOperationException: Not supported by this implementation:
>                 Use EngineSession.settings instead
>         at mozilla.components.browser.engine.servo.ServoEngine.getSettings(ServoEngine.kt:35)
>         at mozilla.components.feature.storage.HistoryTrackingFeature.<init>(HistoryTrackingFeature.kt:20)
>         at org.mozilla.samples.browser.BrowserFragment.onViewCreated(BrowserFragment.kt:42)
>  <...>

--- a/components/browser/engine-servo/src/main/java/mozilla/components/browser/engine/servo/ServoEngine.kt
+++ b/components/browser/engine-servo/src/main/java/mozilla/components/browser/engine/servo/ServoEngine.kt
@@ -11,12 +11,13 @@ import mozilla.components.concept.engine.Engine
 import mozilla.components.concept.engine.EngineSession
 import mozilla.components.concept.engine.EngineView
 import mozilla.components.concept.engine.Settings
+import mozilla.components.concept.engine.history.HistoryTrackingDelegate
 
 /**
  * Servo-based implementation of the Engine interface.
  */
 class ServoEngine(
-    private val defaultSettings: DefaultSettings? = DefaultSettings()
+    private val defaultSettings: DefaultSettings = DefaultSettings()
 ) : Engine {
     override fun createView(context: Context, attrs: AttributeSet?): EngineView {
         return ServoEngineView(context, attrs)
@@ -31,7 +32,13 @@ class ServoEngine(
     /**
      * See [Engine.settings]
      */
-    override val settings: Settings
-        get() = throw UnsupportedOperationException("""Not supported by this implementation:
-            Use EngineSession.settings instead""".trimIndent())
+    override val settings: Settings = object : Settings() {
+        // History tracking is not supported/implemented currently, but we want
+        // to be able to start up with the history tracking feature enabled.
+        override var historyTrackingDelegate: HistoryTrackingDelegate?
+            get() = defaultSettings.historyTrackingDelegate
+            set(value) {
+                defaultSettings.historyTrackingDelegate = value
+            }
+    }
 }

